python/ray/serve/scripts.py

```diff
@@ -220,52 +293,117 @@ def create_deployment(deployment: str, options_json: str):
+        "of the form:\n"
+        '"module.submodule_1...submodule_n.MyClassOrFunction".\n\n'
+        "Blocks after deploying, and logs status periodically. After being killed, "
+        "this command tears down all deployments it deployed. If there are no "
+        "deployments left, it also tears down the Serve application."
+    ),
 )
 @click.argument("config_or_import_path")
 @click.argument("args_and_kwargs", required=False, nargs=-1)
+@click.option(
+    "--runtime-env",
+    type=str,
+    default=None,
+    required=False,
+    help="Path to a local YAML file containing a runtime_env definition. "
+    "Overrides all runtime_envs specified in a config file.",
+)
+@click.option(
+    "--runtime-env-json",
+    type=str,
+    default=None,
+    required=False,
+    help="JSON-serialized runtime_env dictionary. Overrides all runtime_envs "
+    "specified in a config file.",
+)
+@click.option(
+    "--working-dir",
+    type=str,
+    default=None,
+    required=False,
+    help=(
+        "Directory containing files that your job will run in. Can be a "
+        "local directory or a remote URI to a .zip file (S3, GS, HTTP). "
+        "This overrides the working_dir in --runtime-env if both are "
+        "specified. Overrides all working_dirs specified in a config file."
+    ),
+)
+@click.option(
+    "--address",
+    "-a",
+    default="auto",
+    required=False,
+    type=str,
+    help=(
+        'Address of the Ray cluster (not the dashboard) to query. Defaults to "auto".'
+    ),
+)
 def run(
     config_or_import_path: str,
     args_and_kwargs: Tuple[str],
+    runtime_env: str,
+    runtime_env_json: str,
+    working_dir: str,
+    address: str,
 ):
-    """
-    Deploys deployment(s) from CONFIG_OR_IMPORT_PATH, which must be either a
-    Serve YAML configuration file path or an import path to
-    a class or function to deploy. Import paths must be of the form
-    "module.submodule_1...submodule_n.MyClassOrFunction".
-    """
 
     try:
         # Check if path provided is for config or import
         deployments = []
         is_config = pathlib.Path(config_or_import_path).is_file()
         args, kwargs = process_args_and_kwargs(args_and_kwargs)
 
+        # Calculate deployments' runtime env updates requested via args
+        runtime_env_updates = parse_runtime_env_args(
+            runtime_env=runtime_env,
+            runtime_env_json=runtime_env_json,
+        )
+
+        # Get final working_dir
+        working_dir = working_dir or runtime_env_updates.get("working_dir")
+        if "working_dir" in runtime_env_updates:
+            del runtime_env_updates["working_dir"]
+
+        # Create ray.init()'s runtime_env
+        if working_dir is None:
+            ray_runtime_env = {}
+        else:
+            ray_runtime_env = {"working_dir": working_dir}
+            if "working_dir" in runtime_env_updates:
+                del runtime_env_updates["working_dir"]
+
```

Comment:
this logic is a bit hard to follow, could you pull it out into a utility and add some unit tests for it?

Reply:
That logic was needlessly complicated, so I simplified it. Let me know if it's still hard to follow.